addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -298,47 +300,71 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
     _nextQuestion: function () {
         var self = this;
 
-        this.isStartScreen = false;
-        if (this.surveyTimerWidget) {
-            this.surveyTimerWidget.destroy();
-        }
+        this._rpc({
+            // Background management
+            route: "/survey/session/survey_get_next_section_id",
+            params: {
+                'survey_token': this.surveyAccessToken,
+                'current_question_id': this.currentQuestionId ? this.currentQuestionId : 0
+            }
+        }).then(function (nextSection) {
+            var refreshBackgroundPromise = Promise.resolve(false);
+            // Do we need to fade out the background ? Needed here before switching questions to get fade in/out
+            // effect at the same time.
+            if (nextSection.id >= 0) {
+                $('div#wrapwrap').css("box-shadow", "inset 0 0 0 10000px rgba(255,255,255,1)");
+                refreshBackgroundPromise = self._refreshBackground(nextSection.id)
+            }
 
-        var resolveFadeOut;
-        var fadeOutPromise = new Promise(function (resolve, reject) { resolveFadeOut = resolve; });
-        this.$el.fadeOut(this.fadeInOutTime, function () {
-            resolveFadeOut();
-        });
+            self.isStartScreen = false;
+            if (self.surveyTimerWidget) {
+                self.surveyTimerWidget.destroy();
+            }
 
-        var nextQuestionPromise = this._rpc({
-            route: _.str.sprintf('/survey/session/next_question/%s', self.surveyAccessToken)
-        });
+            var resolveFadeOut;
+            var fadeOutPromise = new Promise(function (resolve, reject) { resolveFadeOut = resolve; });
+            self.$el.fadeOut(self.fadeInOutTime, function () {
+                resolveFadeOut();
+            });
 
-        // avoid refreshing results while transitioning
-        if (this.resultsRefreshInterval) {
-            clearInterval(this.resultsRefreshInterval);
-            delete this.resultsRefreshInterval;
-        }
+            var nextQuestionPromise = self._rpc({
+                route: _.str.sprintf('/survey/session/next_question/%s', self.surveyAccessToken)
```

Comment:
Seems like we have a call to survey_get_next_section_id, then session/next_question ? Couldn't we simplify all this code ?